File: OCH_Tao/ABC/C/364.py
N,X,Y = map(int,input().split())
A = list(map(int,input().split()))
B = list(map(int,input().split()))
A.sort(reverse=True)
B.sort(reverse=True)
cntA = 0
cntB = 0
for i in range(N):
  cntA+=A[i]
  cntB+=B[i]
  if cntA>X or cntB>Y:
    print(i+1)
    break
else:
  print(N)

File: OCH_Tao/ABC/C/364.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, X, Y] = input[0].split(" ").map(x => parseInt(x, 10));
const A = input[1].split(" ").map(x => parseInt(x, 10));
const B = input[2].split(" ").map(x => parseInt(x, 10));
A.sort((a, b) => b - a);
B.sort((a, b) => b - a);
let cntA = 0;
let cntB = 0;
for (let i = 0; i < N; i++) {
  cntA += A[i];
  cntB += B[i];
  if (cntA > X || cntB > Y) {
    console.log(i + 1);
    process.exit();
  }
}
console.log(N);